acquisitions/src/main/resources/thunderjet/mod-orders/features/open-order-failure-with-expenditure-restrictions.feature

```diff
@@ -0,0 +1,102 @@
+@parallel=false
+# Created for MODORDERS-1049
+# Related to MODORDERS-528 and open-order-failure-side-effects.feature
+Feature: Open order failure with expenditure restrictions
+
+  Background:
+    * print karate.info.scenarioName
+    * url baseUrl
+    * callonce loginAdmin testAdmin
+    * def okapitokenAdmin = okapitoken
+    * callonce loginRegularUser testUser
+    * def okapitokenUser = okapitoken
+    * def headersAdmin = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitokenAdmin)', 'Accept': 'application/json' }
+    * def headersUser = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitokenUser)', 'Accept': '*/*' }
+    * configure headers = headersUser
+
+    * callonce variables
+
+    * def createOrder = read('classpath:thunderjet/mod-orders/reusable/create-order.feature')
+    * def createOrderLine = read('classpath:thunderjet/mod-orders/reusable/create-order-line.feature')
+
+    * def ledgerId = callonce uuid1
+    * def fundId = callonce uuid2
+    * def budgetId = callonce uuid3
+    * def orderId = callonce uuid4
+    * def poLineId = callonce uuid5
+    * def titleOrPackage = callonce uuid6
+
+
+  Scenario: Create finances
+    * configure headers = headersAdmin
+    * def v = call createLedger { id: "#(ledgerId)", fiscalYearId: "#(globalFiscalYearId)", restrictEncumbrance: false, restrictExpenditures: true }
+    * def v = call createFund { id: "#(fundId)", ledgerId: "#(ledgerId)" }
+    * def v = call createBudget { id: "#(budgetId)", fundId: "#(fundId)", allocated: 5 }
+
+
+  Scenario: Create an order and line
+    * def v = call createOrder { id: "#(orderId)" }
+    * def v = call createOrderLine { id: "#(poLineId)", orderId: "#(orderId)", fundId: "#(fundId)", listUnitPrice: 10, titleOrPackage: titleOrPackage }
+
+
+  Scenario: Try to open the order
+    Given path 'orders/composite-orders', orderId
+    When method GET
+    Then status 200
+
+    * def orderResponse = $
+    * set orderResponse.workflowStatus = 'Open'
+
+    Given path 'orders/composite-orders', orderId
+    And request orderResponse
+    When method PUT
+    Then status 422
+
+
+  Scenario: Check inventory records were not created during the failed open order operation
```

Comment:
inventory records were created initially, but they were rolled backed because of new logic, correct?
Better also to add comment about it, because it can be forgotten in some future

Reply:
That depends. Right now, before MODORDERS-1050 is implemented, we are checking that inventory records were rolled back (which is why I wanted to work on this ticket first). After MODORDERS-1050 is implemented this will simply check that validation prevented the creation of inventory records. Since MODORDERS-1050 is coming up very soon I did not want to mention the rollback in the test.

Reply:
okay got it now, and is it possible to test scenario separately when inventory rollback should be occured?

Reply:
If we had a case this would be a bug (unless we cause some low-level problem in mod-finance-storage, but I am not sure we can rely on that in an integration test). It is safer to have initial validation in mod-oders make sure nothing bad happens before processing. I am not sure the rollback will work well in all cases. For instance suppose you open an order creating inventory automatically, then edit these records a bit, then unopen it without deleting the records, then reopen it and an error happens because of budgets. In this case the rollback would delete the inventory records which is probably not what the user would want.